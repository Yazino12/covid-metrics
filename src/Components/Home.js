import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { BsArrowRightCircle } from 'react-icons/bs';
import Header from './Header';

import { fetchDataApi } from '../Redux/Data/data';

const Home = () => {
  const [searchText, setSearchText] = useState('');
  const data = useSelector((state) => Object.values(state.covidData)[1]);
  let global;
  let countries;

  if (data) {
    [, , global] = Object.values(data);
    [, , , countries] = Object.values(data);
  }
  const filteredCountries = countries?.filter((country) => {
    const newCountry = country.Country.toLowerCase().includes(searchText);
    return newCountry;
  });

  const Navigate = useNavigate();
  const dispatch = useDispatch();
  const condition = true;

  useEffect(() => {
    dispatch(fetchDataApi());
  }, [dispatch]);

  return (
    <>
      <Header conditionProp={condition} />
      <div className="home">
        <div className="header-wrapper">
          <div className="header">{}</div>
          <div className="date-title">
            <h2>{new Date().getFullYear()}</h2>
            <p>COVID UPDATE</p>
          </div>
          <div className="cases-title">
            <h2>{global?.TotalConfirmed.toLocaleString()}</h2>
            <p>CASES WORLD-WIDE</p>
          </div>
        </div>
        <div className="all-stats">
          <div className="search-bar">
            <p className="p">ALL STATS</p>
            <input
              type="text"
              placeholder="Search"
              className="search-bar-input"
              value={searchText}
              onChange={(e) => setSearchText(e.target.value.toLowerCase())}
            />
          </div>
          <div className="stats">
            <div className="stat confirmed">
              <h2>{global?.TotalConfirmed.toLocaleString()}</h2>
              <p>confirmed</p>
            </div>
            <div className="stat recovered">
              <h2>{global?.TotalRecovered.toLocaleString()}</h2>
              <p>recovered</p>
            </div>
            <div className="stat deaths">
              <h2>{global?.TotalDeaths.toLocaleString()}</h2>
              <p>deaths</p>
            </div>
          </div>
        </div>
        <div className="countries">
          {filteredCountries?.map((country) => {
            const countryName = country.Country;
            const countryElement = (
              <div className="country" key={country.ID}>
                <BsArrowRightCircle
                  className="open-details"
                  onClick={() => Navigate(`details/${countryName}`)}
                />
                <h2>{countryName}</h2>
              </div>
            );

            return countryElement;
          })}
        </div>
      </div>
    </>
  );
};

export default Home;
