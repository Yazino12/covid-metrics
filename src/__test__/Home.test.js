import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Home from '../Components/Home';
import store from '../Redux/configStore';

describe('Home tests', () => {
  test('should render', () => {
    const home = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
        ,
      </Provider>
    );

    expect(home).toMatchSnapshot();
  });
});
