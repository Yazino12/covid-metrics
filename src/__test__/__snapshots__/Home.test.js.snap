// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          COVID METRICS
        </h1>
        <div
          class="right"
        >
          <svg
            class="settings"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="3"
            />
            <path
              d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
            />
          </svg>
        </div>
      </header>
      <div
        class="home"
      >
        <div
          class="header-wrapper"
        >
          <div
            class="header"
          />
          <div
            class="date-title"
          >
            <h2>
              2022
            </h2>
            <p>
              COVID UPDATE
            </p>
          </div>
          <div
            class="cases-title"
          >
            <h2 />
            <p>
              CASES WORLD-WIDE
            </p>
          </div>
        </div>
        <div
          class="all-stats"
        >
          <div
            class="search-bar"
          >
            <p
              class="p"
            >
              ALL STATS
            </p>
            <input
              class="search-bar-input"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
          <div
            class="stats"
          >
            <div
              class="stat confirmed"
            >
              <h2 />
              <p>
                confirmed
              </p>
            </div>
            <div
              class="stat recovered"
            >
              <h2 />
              <p>
                recovered
              </p>
            </div>
            <div
              class="stat deaths"
            >
              <h2 />
              <p>
                deaths
              </p>
            </div>
          </div>
        </div>
        <div
          class="countries"
        />
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        COVID METRICS
      </h1>
      <div
        class="right"
      >
        <svg
          class="settings"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </div>
    </header>
    <div
      class="home"
    >
      <div
        class="header-wrapper"
      >
        <div
          class="header"
        />
        <div
          class="date-title"
        >
          <h2>
            2022
          </h2>
          <p>
            COVID UPDATE
          </p>
        </div>
        <div
          class="cases-title"
        >
          <h2 />
          <p>
            CASES WORLD-WIDE
          </p>
        </div>
      </div>
      <div
        class="all-stats"
      >
        <div
          class="search-bar"
        >
          <p
            class="p"
          >
            ALL STATS
          </p>
          <input
            class="search-bar-input"
            placeholder="Search"
            type="text"
            value=""
          />
        </div>
        <div
          class="stats"
        >
          <div
            class="stat confirmed"
          >
            <h2 />
            <p>
              confirmed
            </p>
          </div>
          <div
            class="stat recovered"
          >
            <h2 />
            <p>
              recovered
            </p>
          </div>
          <div
            class="stat deaths"
          >
            <h2 />
            <p>
              deaths
            </p>
          </div>
        </div>
      </div>
      <div
        class="countries"
      />
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
